src/reanimated2/Hooks.js

```diff
@@ -249,25 +255,40 @@ function styleUpdater(viewTag, updater, state) {
   }
 }
 
-export function useAnimatedStyle(updater) {
+export function useAnimatedStyle(updater, dependencies) {
   const viewTag = useSharedValue(-1);
-
   const initRef = useRef(null);
+  const inputs = Object.values(updater._closure);
+
+  // build dependencies
+  if (dependencies === undefined) {
+    dependencies = [...inputs, updater.__workletHash];
```

Comment:
Why don't we just pass undefined to useEffect?

Reply:
Do you want users to use [] explicitly if they want it to not recreate?

Reply:
Currently, you have to pass null if you don't want to recreate worklet. I just think that there is no need to add an input array to dependencies in any case. It won't work with complex objects. For instance:
```JS
useAnimatedStyle(() => {return {backgroundColor: state.color}});
```
In my option, the default behavior should be recreating always or never.

Reply:
This has been discused beyond github